pkg/server/catalog/catalog_test.go

```diff
@@ -0,0 +1,252 @@
+	nr_noop "github.com/spiffe/spire/pkg/server/plugin/noderesolver/noop"
+	"github.com/spiffe/spire/pkg/server/plugin/upstreamauthority"
+	"github.com/spiffe/spire/pkg/server/plugin/upstreamca"
+	spi "github.com/spiffe/spire/proto/spire/common/plugin"
+	"github.com/spiffe/spire/test/fakes/fakeagentstore"
+	"github.com/spiffe/spire/test/fakes/fakedatastore"
+	"github.com/spiffe/spire/test/fakes/fakeidentityprovider"
+	"github.com/spiffe/spire/test/fakes/fakemetrics"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestLoad(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	// Create a configuration with minimal required plugins
+	hclConfig := createDefaultConfig()
+
+	// Create a custom list of BuiltIn plugins, it contains the minimal necessary plugins, and fakes for Datastore, UpstreamCA and UpstreamAuthority
+	builtIn = []catalog.Plugin{
+		// Fake Datastore to prevent sql connections
+		catalog.MakePlugin("sql",
+			datastore.PluginServer(fakedatastore.New())),
+		na_join_token.BuiltIn(),
+		km_memory.BuiltIn(),
+		nr_noop.BuiltIn(),
+		// Fake UpstreamCA
+		catalog.MakePlugin("testUpstream",
+			upstreamca.PluginServer(fakeUpstreamCAPlugin{})),
+		// Fake UpstreamAuthority
+		catalog.MakePlugin("testUpstream",
+			upstreamauthority.PluginServer(fakeUpstreamAuthorityPlugin{})),
+	}
+
+	// Create all fakes needed on Load method
+	identityProvider := fakeidentityprovider.New()
+	agentStore := fakeagentstore.New()
+
+	metricsService := metricsservice.New(metricsservice.Config{
+		Metrics: fakemetrics.New(),
+	})
+	log, _ := test.NewNullLogger()
+
+	testCases := []struct {
+		// Test case name
+		name string
+		// Function to create plugin configurations
+		createHclConfig func() HCLPluginConfigMap
+		// Ported UpstreamCAs
+		ported map[string]bool
+		// Expected error
+		err string
+		// Method with validations for repository response
+		verifyRepo func(*testing.T, *Repository)
+	}{
+		{
+			name: "no UpstreamCA or UpstreamAuthority config",
+			createHclConfig: func() HCLPluginConfigMap {
+				return hclConfig
+			},
+			verifyRepo: func(t *testing.T, repository *Repository) {
+				// No UpstreamAuthority expected
+				upstream, ok := repository.Catalog.GetUpstreamAuthority()
+				require.False(t, ok)
+				require.Nil(t, upstream)
+			},
+		},
+		{
+			name: "unported UpstreamCA",
+			createHclConfig: func() HCLPluginConfigMap {
+				c := hclConfig
+				// Add fakeUpstreamCA to configuration
+				c[upstreamca.Type] = map[string]HCLPluginConfig{"testUpstream": {}}
+				return c
+			},
+			verifyRepo: func(t *testing.T, repository *Repository) {
+				// An upstreamAuthority expected and it is calling fakeUpstreamCA method
+				upstream, ok := repository.Catalog.GetUpstreamAuthority()
+				require.True(t, ok)
+				require.NotNil(t, upstream)
+
+				_, err := upstream.MintX509CA(ctx, &upstreamauthority.MintX509CARequest{})
```

Comment:
just assert that `upstream` isn't a wrapper type... no need to invoke a method here